src/main/java/gift/controller/ProductController.java

```diff
@@ -24,13 +34,23 @@ public class ProductController {
 
     private final ProductService productService;
 
+    private final String key = "Yn2kjibddFAWtnPJ2AFlL8WXmohJMCvigQggaEypa5E";
+
     public ProductController(ProductService productService) {
         this.productService = productService;
     }
 
     @PostMapping
     public ResponseEntity<ProductResponseDto> createProduct(
-            @Valid @RequestBody CreateProductRequestDto requestDto) {
+            @Valid @RequestBody CreateProductRequestDto requestDto,
+            @RequestHeader(value = "Authorization") String token) {
+        Optional<Member> optionalMember = isValidateToken(token);
+        if (optionalMember.isEmpty()) {
+            throw new CustomException(ErrorCode.NotLogin);
+        }
```

Comment:
```suggestion
        Member optionalMember = isValidateToken(token).orElseThrow(() -> new CustomException(ErrorCode.NotLogin));
```
Optional의 메서드를 활용해서 한 줄로도 표현 가능합니다! 이건 코드 스타일의 차이니 참고만 해주세요 ㅎㅎ